### PR TITLE
split missed schedule metrics with back execution info

### PR DESCRIPTION
**Pull request by ShellyWEI**
- Reveal the missed schedule count with back execution feature enabled or not.
- Reveal the count of back execution enabled triggers.

### CODE REVIEW

File: azkaban-common/src/main/java/azkaban/trigger/TriggerManager.java [hunk 1]
```diff
@@ -52,6 +52,7 @@ public class TriggerManager extends EventHandler implements TriggerManagerAdapte
   private static final Logger logger = Logger.getLogger(TriggerManager.class);
   private static final Map<Integer, Trigger> triggerIdMap = new ConcurrentHashMap<>();
   private static final Set<Integer> removedTriggerIds = new HashSet<>();
+  private static final Set<Integer> backExecuteEnabledTriggerIds = new HashSet<>();
```
Comment (byjiang1996) [inline azkaban-common/src/main/java/azkaban/trigger/TriggerManager.java:55]: I'm surprised all the Sets in this class are not "ConcurrentHashSet". The data structure is not thread-safe.

File: azkaban-common/src/main/java/azkaban/trigger/TriggerManager.java [hunk 2]
```diff
@@ -148,6 +156,11 @@ public void updateTrigger(final Trigger t) throws TriggerManagerException {
     this.runnerThread.deleteTrigger(triggerIdMap.get(t.getTriggerId()));
     this.runnerThread.addTrigger(t);
     triggerIdMap.put(t.getTriggerId(), t);
+    if (t.isBackExecuteOnceOnMiss()) {
+      backExecuteEnabledTriggerIds.add(t.getTriggerId());
+    } else {
+      backExecuteEnabledTriggerIds.remove(t.getTriggerId());
```
Comment (byjiang1996) [inline azkaban-common/src/main/java/azkaban/trigger/TriggerManager.java:162]: Do you need to run
"if backExecuteEnabledTriggerIds.containsKey(t.getTriggerId())" before removing it?

File: azkaban-common/src/main/java/azkaban/trigger/TriggerManager.java [hunk 3]
```diff
@@ -164,6 +177,7 @@ public void removeTrigger(final Trigger t) throws TriggerManagerException {
     this.runnerThread.deleteTrigger(t);
     triggerIdMap.remove(t.getTriggerId());
     removedTriggerIds.add(t.getTriggerId());
+    backExecuteEnabledTriggerIds.remove(t.getTriggerId());
```
Comment (byjiang1996) [inline azkaban-common/src/main/java/azkaban/trigger/TriggerManager.java:180]: same above.

File: azkaban-common/src/main/java/azkaban/trigger/TriggerManager.java (hunk 2 above)
Comment (ShellyWEI) [reply azkaban-common/src/main/java/azkaban/trigger/TriggerManager.java:162]: This is a boolean method, if the value is not present, it simply return false but I don't care the return value.

File: azkaban-common/src/main/java/azkaban/trigger/TriggerManager.java (hunk 1 above)
Comment (byjiang1996) [reply azkaban-common/src/main/java/azkaban/trigger/TriggerManager.java:55]: Oh, i see it now. Thanks for your explanation!

File: azkaban-common/src/main/java/azkaban/trigger/TriggerManager.java (hunk 2 above)
Comment (byjiang1996) [reply azkaban-common/src/main/java/azkaban/trigger/TriggerManager.java:162]: Interesting. Thanks for letting me know.

**Review by byjiang1996 — APPROVED** [review]

**Pull request merged**